packages/thirdweb/src/extensions/erc721/drops/write/updateMetadata.ts

```diff
@@ -0,0 +1,93 @@
+import type { ThirdwebClient } from "../../../../client/client.js";
+import type { BaseTransactionOptions } from "../../../../transaction/types.js";
+import type { NFT, NFTInput } from "../../../../utils/nft/parseNft.js";
+import { getBaseURICount } from "../../__generated__/IBatchMintMetadata/read/getBaseURICount.js";
+
+export type UpdateMetadataParams = {
+  targetTokenId: bigint;
+  newMetadata: NFTInput;
+  client: ThirdwebClient;
+};
+
+export async function updateMetadata(
```

Comment:
this function should only 'prepare' the transaction so it should not be `async`

instead, all the work should happen inside the `params` async function (yes it can be a function!)

the idea is that you defer the heavy execution to when the user actually tries to send the transaction, making 'preparing' this transaction very cheap